### ISSUE TITLE
A bug in albumentations 0.4.6 when use it in a custom Keras Generator

### ISSUE THREAD

**Issue by art23130** [albumentations-team/albumentations#846]
## 🐛 Bug

Thanks for your nice plugin, we use it a lot in our research.

I have found a bug in albumentations 0.4.6 that I thought it is better to share it with you.
The bug I am reporting is technically make albumentations unusual in model developmentusing tensorflow.

When we use albumentations.ReplayCompose or albumentations.Compose inside a custom generator (tf.keras.utils.Sequence), albumentation applies different compose to image and mask. So the outcomes are not match any more.

## To Reproduce

Steps to reproduce the behaviour:

1. Create a keras generator using tf.keras.utils.Sequence (I have attached the code I have developed)

## Environment

 - Albumentations version: 0.4.6
 - Python version: 3.7.9
 - OS: Windows 10
 - tensorflow: 2.4.0
 - How you installed albumentations (`conda`, `pip`, source): conda install 
 - Any other relevant information:

Like to the code: https://gist.github.com/art23130/bab7307daf7b63b0d713d69c654c628b)

This is how you use the above code:

generator4training = MasterGenerator4SS(
    image_filenames = image_train,
    label_list = label_train,
    nb_bands = band_no, 
    batch_size = 10,
    image_size = image_size,
    nb_classes = class_no,
    augmentation = augment_image_labels,
    random_shuffle = False)

Where augment_image_labels is the function as below:

```
def augment_image_labels(_image, _mask):
    transform = albumentations.Compose([
        albumentations.HorizontalFlip(p=0.7),
        albumentations.VerticalFlip(p=0.7),
        albumentations.RandomRotate90(p=0.7),
        albumentations.Transpose(p=0.7)],p=0.9)

    augmented = transform(image = _image, mask = _mask)

    return augmented
```

**Comment by creafz** [comment]
Hey @art23130 

It is difficult to debug this issue without access to the input data. I wrote a simple script, that passes values to `augment_image_labels` and checks whether they were augmented in the same way, and it works fine with the latest version of Alumentations (0.5.2). At a first glance, I cant see why Keras Generator could make those values incorrect.

```
image = np.repeat(np.eye(16, 16)[:, :, np.newaxis], repeats=4, axis=2).astype(np.float32)
mask = np.copy(image[:, :, 0]).astype(np.int8)
for iter_ in range(100):
    augmented = augment_image_labels(image, mask)
    augmented_image = augmented['image']
    augmented_mask = augmented['mask']
    for channel in range(4):
        assert np.array_equal(augmented_image[:, :, channel], augmented_mask)
```

A few ideas:
1. Could you try to update Albumentations to the latest version (0.4.6 -> 0.5.2) and see whether the problem still occurs?
2. On [line 72](https://gist.github.com/art23130/bab7307daf7b63b0d713d69c654c628b#file-keras_generator-py-L72) the generator loads labels as `np.int8` (signed int8), but on [line 100](https://gist.github.com/art23130/bab7307daf7b63b0d713d69c654c628b#file-keras_generator-py-L100) it uses the `np.uint8` datatype (unsigned int8). May be there is a problem with negative integer values.

**Comment by art23130** [comment]
Hey @creafz 
We have tried different approach to solve this issue but still no progress. 
We have also tried your suggestion an found out that Albumentations works fine, the issue raises when generator returns the images.
We think that there is a conflict between Albumentation and this Keras Module.

Actually, we have diverted our workflow to tf.data to be able to use Albumentations.
